internal/runtimehandlerhooks/utils_linux.go

```diff
@@ -160,41 +149,47 @@ func isServiceEnabled(serviceName string) bool {
 	return false
 }
 
-func updateIrqBalanceConfigFile(irqBalanceConfigFile, newIRQBalanceSetting string) error {
+func updateIrqBalanceConfigFile(irqBalanceConfigFile string, newIRQBalanceSetting cpuset.CPUSet) error {
 	input, err := os.ReadFile(irqBalanceConfigFile)
 	if err != nil {
 		return err
 	}
 	lines := strings.Split(string(input), "\n")
 	found := false
 	for i, line := range lines {
+		// Comment out the old deprecated variable
 		if strings.HasPrefix(line, irqBalanceBannedCpus+"=") {
-			lines[i] = irqBalanceBannedCpus + "=" + "\"" + newIRQBalanceSetting + "\""
+			lines[i] = "#" + line
+		}
+
+		if strings.HasPrefix(line, irqBalanceBannedCPUList+"=") {
+			lines[i] = irqBalanceBannedCPUList + "=" + "\"" + newIRQBalanceSetting.String() + "\""
 			found = true
 		}
 	}
 	output := strings.Join(lines, "\n")
 	if !found {
-		output = output + "\n" + irqBalanceBannedCpus + "=" + "\"" + newIRQBalanceSetting + "\"" + "\n"
+		output = output + "\n" + irqBalanceBannedCPUList + "=" + "\"" + newIRQBalanceSetting.String() + "\"" + "\n"
 	}
 	if err := os.WriteFile(irqBalanceConfigFile, []byte(output), 0o644); err != nil {
 		return err
 	}
 	return nil
 }
 
-func retrieveIrqBannedCPUMasks(irqBalanceConfigFile string) (string, error) {
+func retrieveIrqBannedCPUList(irqBalanceConfigFile string) (cpuset.CPUSet, error) {
 	input, err := os.ReadFile(irqBalanceConfigFile)
 	if err != nil {
-		return "", err
+		return cpuset.New(), err
 	}
 	lines := strings.Split(string(input), "\n")
 	for _, line := range lines {
-		if strings.HasPrefix(line, irqBalanceBannedCpus+"=") {
```

Comment:
I would read both to allow for seamless transition. You are commenting out the old variable above anyway.

Reply:
we definitely can read the legacy value, but then what shall we be doing with it?

Reply:
Merge with the new one and then comment it out?

Reply:
Done. I added the logic and a comment that explains the new flow